### PR TITLE
TandemFoil: weight decay 1e-2 — faster convergence under cold-start

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

Haku's WD sweep showed WD=1e-2 converges significantly faster than the default WD=1e-4 with Lion optimizer at the same epoch count (val MAE 95.59 vs 103.87 at 7 epochs). The TandemFoil dataset has a severe cold-start problem that limits training to 7-8 epochs per 30-min run. If WD=1e-2 continues this faster convergence pattern at our current best configuration (T_max=10, all physics features enabled), it may reach or beat the 75.59 baseline even with fewer epochs.

Stronger regularization (WD=1e-2) may also help generalization on the OOD evaluation splits (geom_camber_rc, geom_camber_cruise, re_rand).

## Baseline to Beat

- **Primary metric:** `val_primary/surface_pressure_mae`
- **Current best:** 75.59 val / 72.12 test (PR #2490, T_max=10, Lion lr=3e-4, 14 epochs)
- **W&B run:** 77yoba65
- **Reproduce baseline:** `cd target/icml2026 && python train.py --dataset tandemfoil --optimizer lion --lr 3e-4 --cosine-t-max 10 --no-use-ema --model-slices 64 --enable-fourier --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition --epochs 999`

## Instructions

Run the following command. Training will run until the 30-min SENPAI timeout (~7-8 epochs). Use W&B to track val_primary/surface_pressure_mae per epoch.

```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset tandemfoil --optimizer lion --lr 3e-4 --cosine-t-max 10 \
  --no-use-ema --model-slices 64 --enable-fourier \
  --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only \
  --asinh-pressure --residual-prediction --enable-pressure-prior-addition \
  --weight-decay 1e-2 --epochs 999 \
  --agent violet --wandb-name "violet/tandem-wd1e2" \
  --wandb-group "tandem-tmax10-wd-sweep"
```

**W&B group:** `tandem-tmax10-wd-sweep`

## What to Report

- Best `val_primary/surface_pressure_mae` (check all epochs in W&B)
- Per-split val MAE: single_in_dist, geom_camber_rc, geom_camber_cruise, re_rand
- Total epochs completed before timeout
- Whether convergence was faster/slower than baseline
- W&B run ID

## Context: WD Sweep Prior Results (Haku, different config)

| WD    | val MAE at epoch 7 | Notes                  |
|-------|--------------------|------------------------|
| 1e-4  | 103.87             | default                |
| 1e-2  | 95.59              | -7.9% vs default WD    |

Note: this was at a slightly different config. We're testing WD=1e-2 at the full best config (all physics features + Fourier + T_max=10).

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT violet:

## Results

### WD=1e-2 Experiment (TandemFoil, full best config)

| Metric | WD=1e-2 (this run) | Baseline (WD=1e-4, PR #2490) | Delta |
|--------|-------------------|-------------------------------|-------|
| **val_primary/surface_pressure_mae** | **101.08** (best, epoch 6) | **75.59** (epoch 14) | +33.7% worse |
| test_primary/surface_pressure_mae | 103.16 (final epoch) | 72.12 | +43.0% worse |
| Epochs completed | 7 (30-min timeout) | 14 | -50% fewer epochs |

### Per-split val MAE at best epoch (epoch 6)

| Split | WD=1e-2 | Baseline (epoch 14) |
|-------|---------|---------------------|
| single_in_dist | 117.04 | — |
| geom_camber_rc | 110.68 | — |
| geom_camber_cruise | 82.92 | — |
| re_rand | 93.69 | — |

### Per-split test MAE (final epoch 7)

| Split | WD=1e-2 |
|-------|---------|
| single_in_dist | 100.70 |
| geom_camber_rc | 103.03 |
| geom_camber_cruise | 105.11 |
| re_rand | 103.80 |

### Epoch-by-epoch val_primary/surface_pressure_mae

| Epoch | val MAE | LR |
|-------|---------|-----|
| 1 | 204.54 | 0.0 |
| 2 | 154.94 | 3e-4 |
| 3 | 120.11 | 0.0 |
| 4 | 115.06 | 3e-4 |
| 5 | 105.48 | 0.0 |
| 6 | **101.08** | 3e-4 |
| 7 | 106.52 | 0.0 |

### Experiment details

- **Command:** `cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py --dataset tandemfoil --optimizer lion --lr 3e-4 --cosine-t-max 10 --no-use-ema --model-slices 64 --enable-fourier --enable-te-coord-frame --enable-cp-panel --enable-cp-panel-tandem-only --asinh-pressure --residual-prediction --enable-pressure-prior-addition --weight-decay 1e-2 --epochs 999 --agent violet --wandb-name "violet/tandem-wd1e2" --wandb-group "tandem-tmax10-wd-sweep"`
- **W&B run:** dlaf9w6c
- **W&B group:** tandem-tmax10-wd-sweep
- **Peak memory:** Not captured in logs (GPU freed after training)
- **Epochs:** 7 (30-min SENPAI timeout)

### What happened

**Negative result.** WD=1e-2 did NOT beat the baseline at the full best config.

- Best val MAE was 101.08 at epoch 6, far from the 75.59 baseline (which had 14 epochs).
- The run only completed 7 epochs vs 14 for the baseline. This is the same epoch count as Haku's earlier comparison, but the per-epoch time is identical — the difference is just the 30-min timeout cutting in at 7 epochs (vs 14 epochs for the baseline which may have had a longer budget or faster epochs).
- Comparing at equal epoch counts is difficult without the baseline's epoch-by-epoch trajectory, but 101.08 at epoch 6-7 is substantially above 75.59 at epoch 14. Even assuming linear improvement, WD=1e-2 would need significantly more epochs to approach baseline.
- The oscillation pattern (odd epochs at LR=0 tend worse, even epochs at LR=3e-4 tend better) is consistent with T_max=10 cosine cycling.
- Haku's prior results showed WD=1e-2 at 95.59 vs WD=1e-4 at 103.87 at epoch 7 — suggesting WD=1e-2 converges faster *at Haku's config*. Our result of 101.08 at epoch 6 is in a similar range, but the full best config (all physics features + Fourier) may already provide sufficient regularization, reducing the marginal benefit of stronger WD.

### Suggested follow-ups

1. **Head-to-head comparison at matched epochs:** Run the default WD=1e-4 baseline for exactly 7 epochs under the same timeout to get a fair convergence-speed comparison.
2. **Intermediate WD values:** Try WD=5e-3 or WD=3e-3 — the jump from 1e-4 to 1e-2 is 100x; a gentler increase might find the sweet spot.
3. **WD=1e-2 with longer training:** If the baseline's 14-epoch run had a 60-min budget, testing WD=1e-2 with the same extended budget would reveal whether it catches up or plateaus.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2675 because ADVISOR: Closing — TandemFoil WD=1e-2 reached 101.08 at 7 epochs (33% worse than 75.59 baseline at 14 ep). Cold-start problem means only 7 epochs, making comparison impossible. Direction not productive at current config..
